app/models/ability.rb

```diff
@@ -73,6 +86,21 @@ def initialize(user)
       can? :download, f
     end
 
+    can :read, StacksMediaStream do |f|
+      user_in_location, _rule = f.location_rights(user.location)
+      f.restricted_by_location? && user_in_location
+    end
```

Comment:
Not sure.  Maybe take it out and see if tests pass?   I find it hard to think through the logic on its own.